helix-term/tests/test/commands.rs

```diff
@@ -131,3 +131,29 @@ async fn test_selection_duplication() -> anyhow::Result<()> {
     .await?;
     Ok(())
 }
+
+#[tokio::test]
+async fn increment_integer_test() -> anyhow::Result<()> {
+    // correct selection placement with multiple -1 to zero
+    // `-1-1-2-1-1` becomes `00-100`
+    test((
+        "#[-|]#1#(-|)#1#(-|)#2#(-|)#1#(-|)#1",
+        "<C-a>",
+        "#[0|]##(0|)##(-|)#1#(0|)##(0|)#",
+    ))
+    .await?;
+
+    // correct selection placement with multiple -1 to zero, incremented twice
+    // `-1-1-2-1-1` becomes `00-100` then `01-99`
+    test((
+        "#[-|]#1#(-|)#1#(-|)#2#(-|)#1#(-|)#1",
+        "<C-a><C-a>",
+        "#[0|]#1#(-|)#99",
+    ))
+    .await?;
```

Comment:
I understand what's happening here with the numbers colliding but I think this behavior is confusing: I wouldn't expect incrementing or decrementing to change the number of selections. If I have 5 selections over the negatives, I would expect to be able to retain those selections as I increment/decrement and always end up with 5 selections

I'm not sure it's possible to have the behavior I expect without separating the navigation part of C-a/C-x from the incrementing/decrementing behavior (https://github.com/helix-editor/helix/issues/3028)